### PR TITLE
Add a setting to let user to choose coreaudio or avfoundation as the audio driver

### PR DESCRIPTION
**Pull request by uiryuu**
- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #5064.

---

**Description:**
Added descriptive text to indicate AVFoundation is an experimental audio driver which may cause issues.

image

### CODE REVIEW

File: iina/MPVController.swift [hunk 1]
```diff
@@ -517,8 +517,9 @@ class MPVController: NSObject {
 
     // As mpv support for audio using the AVFoundation framework is new we enable it before applying
     // user's settings. This allows a user to roll back to the Core Audio framework should a problem
-    // be encountered with the new code.
-    chkErr(setOptionString(MPVOption.Audio.ao, "avfoundation", level: .verbose))
+    // be encountered with the new code. Set audio driver based on user settings.
+    let driver = Preference.bool(for: .audioDriverEnableAVFoundation) ? "avfoundation" : "coreaudio"
+    chkErr(setOptionString(MPVOption.Audio.ao, driver, level: .verbose))
```
Comment (low-batt) [inline iina/MPVController.swift]:
```suggestion
    setUserOption(PK.audioDriverEnableAVFoundation, type: .other, forName: MPVOption.Audio.ao,
                  level: .verbose) { key in
      Preference.bool(for: key) ? "avfoundation" : "coreaudio"
    }
```

File: iina/MPVController.swift [hunk 2]
```diff
@@ -517,8 +517,9 @@ class MPVController: NSObject {
 
     // As mpv support for audio using the AVFoundation framework is new we enable it before applying
     // user's settings. This allows a user to roll back to the Core Audio framework should a problem
```
Comment (low-batt) [inline iina/MPVController.swift]: This comment was about the user being able to add "ao=coreaudio" in advanced settings to rollback from AVFoundation to Core Audio. So the comment is a bit out of date now that we have an explicit setting for this.

**Review by low-batt — COMMENTED** [review]
Changes look good. With the suggested code change the audio driver can be dynamically changed. There is a couple of seconds delay and then mpv starts the audio again using the selected driver. This might be useful to users who like Spatial Audio, but don't like the mute delay bug with AVFoundation. So they might switch to AVFoundation only when playing a video with Spatial Audio. I'm thinking make this dynamic, just because I think apps should always avoid the requirement to restart after changing a setting, if possible. If we do make it dynamic we should update the Inspector window to understand the driver can change. Thoughts?

**Comment by low-batt** [comment]
Noticed one other thing. The help button is way over to the right edge whereas other help buttons are close to the setting.

**Comment by uiryuu** [comment]
> The help button is way over to the right edge whereas other help buttons are close to the setting.

I intentionally did this, because I also added a "flask" icon to indicate AVFoundation is an experimental option. I moved the question mark away to separate those 2 icons.

**Comment by low-batt** [comment]
This was included in IINA 1.4.0.

**Pull request merged**